app/redux/shared_actions/SetsActionCreators.js

```diff
@@ -96,3 +135,22 @@ export const finishedUploadingSets = (revision) => ({
 });
 
 export const failedUploadSets = () => ({ type: FAILED_UPLOAD_SETS });
+
+const logAnalytics = (set, auto_end_timer, has_reps, is_previous_set_fields_filled, num_fields_entered, is_default_end_timer, manuallyStarted, wasSanityCheck, previous_set_has_reps, state) => {
```

Comment:
instead of passing params in, IMO pass in the state and have this function pull what it needs for the analytics

Reply:
Make all of these state?